packages/engine/tests/simulations/mod.rs

```diff
@@ -72,7 +72,28 @@ macro_rules! run_test {
 }
 
 mod state {
```

Comment:
Commenting here just so that we can keep discussion inline (Github is annoying and doesn't let you properly reply to top-level comments)

I think we need to test for multiple agents, I'm unsure of the value of doing so for every test, we might need to think about that (@litvand thoughts?), but I know in the past we've had state get borked where we thought it was working but actually only the first agent could read and write state, and the others silently failed or just had null (or maybe default) values.

Perhaps that'd be fine as its own test as I don't think the data-types should really matter here.

Reply:
I'd prefer different agents just in case we somehow send the same segment to multiple accidentally or something. But I think we probably only need one or two tests to specifically test multiple agents, rather than further complicating the existing tests with multiple agents for each scenario.

Reply:
I added multiple agents to the `number` and `string` tests

Reply:
I do wonder if we should have made a new test just so we would see a test for explicitly checking multiple agents, but I think this is fine